### DISCUSSION TITLE
Add support for `--env-file=` flag for node 20 in `nodeArguments`

### DISCUSSION THREAD

**Discussion by torbjornvatn** [avajs/ava#3293] [Q&A]
Please provide details about:

* What you're trying to do
I was hoping that I could add a config like:

```json
"nodeArguments": [
   "--env-file=.env"
],
```
and get access to the environments loaded by Node from that file.
[--env-fileconfig](https://nodejs.org/api/cli.html#--env-fileconfig)

* Why you can't use AVA for this
Even though ava accepts the nodeArguments config above the result of running:
```javascript
console.log(process.execArgv)
console.log(process.version)
console.log(process.env.DB_CONNECTION)
```
in a test is this:
```
[ '--env-file=.env', '--enable-source-maps' ]
v20.10.0
undefined
```

* And maybe how you think AVA could handle this
Would it be possible to make ava pass this on to node so it gets loaded as environment variables

**Comment by novemberborn** [comment]
`nodeArguments` are passed when creating worker threads (default) or child processes.

However with worker threads, not all arguments have an effect. Some values are inherited from the main process. My guess is that this is the case for `--env-file`.

Running the following should work:

```console
NODE_OPTIONS="--env-file=.env" npx ava
```

This way, the env vars are added to the main AVA process and should be inherited by the worker threads.

See also https://github.com/nodejs/node/issues/37410.

**Comment by torbjornvatn** [comment]
Hmm, seems like that's not allowed actually 
> node:  --env-file= is not allowed in NODE_OPTIONS
It's not in this [list](https://nodejs.org/docs/latest-v20.x/api/cli.html#node_optionsoptions)

I've solved it by falling back to `dotenv` for now, but hopefully the worker threads will get support for this eventually.

Could it be an alternative to add an `envFile` config to Ava as a supplement to `environmentVariables`?

**Reply by novemberborn** [reply]
A good starting point would be to use `dotenv` in the `ava.config.js` and set `environmentVariables` to the resulting object.
